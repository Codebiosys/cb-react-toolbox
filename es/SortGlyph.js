'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _lodash = require('lodash');

var _reactBootstrap = require('react-bootstrap');

require('bootstrap/dist/css/bootstrap.css');

require('react-datetime/css/react-datetime.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SortGlyph = function SortGlyph(_ref) {
  var field = _ref.field,
      orderBy = _ref.orderBy,
      type = _ref.type,
      reverse = _ref.reverse,
      changeSortOrder = _ref.changeSortOrder,
      reverseSortOrder = _ref.reverseSortOrder;

  var updateSort = function updateSort() {
    if ((0, _lodash.toUpper)(field) === (0, _lodash.toUpper)(orderBy)) {
      reverseSortOrder(reverse);
    } else {
      changeSortOrder(field);
    }
  };
  return _react2.default.createElement(_reactBootstrap.Glyphicon, {
    onClick: function onClick() {
      return updateSort();
    },
    glyph: (0, _lodash.toUpper)(field) === (0, _lodash.toUpper)(orderBy) && reverse ? 'sort-by-' + type + '-alt' : 'sort-by-' + type,
    className: (0, _lodash.toUpper)(field) === (0, _lodash.toUpper)(orderBy) ? 'text-success' : ''
  });
};

SortGlyph.propTypes = {
  field: _propTypes2.default.string.isRequired,
  orderBy: _propTypes2.default.string,
  type: _propTypes2.default.string,
  reverse: _propTypes2.default.bool,
  changeSortOrder: _propTypes2.default.func.isRequired,
  reverseSortOrder: _propTypes2.default.func.isRequired
};

SortGlyph.defaultProps = {
  orderBy: '',
  reverse: false,
  type: 'alphabet'
};

exports.default = SortGlyph;