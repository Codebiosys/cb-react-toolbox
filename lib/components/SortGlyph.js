'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _lodash = require('lodash');

var _reactBootstrap = require('react-bootstrap');

require('bootstrap/dist/css/bootstrap.css');

require('react-datetime/css/react-datetime.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SortGlyph = function SortGlyph(_ref) {
  var field = _ref.field,
      orderBy = _ref.orderBy,
      type = _ref.type,
      reverse = _ref.reverse,
      changeSortOrder = _ref.changeSortOrder,
      reverseSortOrder = _ref.reverseSortOrder;

  var updateSort = function updateSort() {
    if ((0, _lodash.toUpper)(field) === (0, _lodash.toUpper)(orderBy)) {
      reverseSortOrder(reverse);
    } else {
      changeSortOrder(field);
    }
  };
  return _react2.default.createElement(_reactBootstrap.Glyphicon, {
    onClick: function onClick() {
      return updateSort();
    },
    glyph: (0, _lodash.toUpper)(field) === (0, _lodash.toUpper)(orderBy) && reverse ? 'sort-by-' + type + '-alt' : 'sort-by-' + type,
    className: (0, _lodash.toUpper)(field) === (0, _lodash.toUpper)(orderBy) ? 'text-success' : ''
  });
};

SortGlyph.defaultProps = {
  orderBy: '',
  reverse: false,
  type: 'alphabet'
};

exports.default = SortGlyph;