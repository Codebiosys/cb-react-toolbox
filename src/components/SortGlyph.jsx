import React from 'react';
import PropTypes from 'prop-types';
import { toUpper } from 'lodash';
import {
  Glyphicon,
} from 'react-bootstrap';

import 'react-datetime/css/react-datetime.css';


const SortGlyph = ({
  field,
  orderBy,
  type,
  reverse,
  changeSortOrder,
  reverseSortOrder,
}) => {
  const updateSort = () => {
    if (toUpper(field) === toUpper(orderBy)) {
      reverseSortOrder(reverse);
    } else {
      changeSortOrder(field);
    }
  };
  return (
    <Glyphicon
      onClick={() => updateSort()}
      glyph={toUpper(field) === toUpper(orderBy) && reverse ? `sort-by-${type}-alt` : `sort-by-${type}`}
      className={toUpper(field) === toUpper(orderBy) ? 'text-success' : ''}
    />
  );
};

SortGlyph.propTypes = {
  field: PropTypes.string.isRequired,
  orderBy: PropTypes.string,
  type: PropTypes.string,
  reverse: PropTypes.bool,
  changeSortOrder: PropTypes.func.isRequired,
  reverseSortOrder: PropTypes.func.isRequired,
};

SortGlyph.defaultProps = {
  orderBy: '',
  reverse: false,
  type: 'alphabet',
};

export default SortGlyph;
