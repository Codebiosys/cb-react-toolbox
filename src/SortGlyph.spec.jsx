import React from 'react';
import { shallow } from 'enzyme';

import SortGlyph from './SortGlyph';

const changeSortOrder = jest.fn();
const reverseSortOrder = jest.fn();

const componentProps = {
  field: 'test',
  orderBy: 'notTest',
  type: 'alphabet',
  reverse: false,
  changeSortOrder,
  reverseSortOrder,
};


describe('The Sort Glyph', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SortGlyph {...componentProps} />,
    );
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('supports alternative sort glyphs', () => {
    const altProps = { ...componentProps, type: 'test' };
    const altWrapper = shallow(
      <SortGlyph {...altProps} />,
    );
    expect(altWrapper.prop('glyph')).toEqual(`sort-by-${altProps.type}`);
  });

  it('Highlights the glyph when the orderBy matches the field', () => {
    const matchingProps = { ...componentProps, orderBy: 'test' };
    const matchedwrapper = shallow(
      <SortGlyph {...matchingProps} />,
    );
    expect(matchedwrapper.prop('className')).toEqual('text-success');
  });

  it('Displays the alt glyph when the orderBy matches the field and reverse is true', () => {
    const reversedProps = { ...componentProps, orderBy: 'test', reverse: true };
    const reversedWrapper = shallow(
      <SortGlyph {...reversedProps} />,
    );
    expect(reversedWrapper.prop('glyph')).toEqual(`sort-by-${reversedProps.type}-alt`);
  });
  it('Orders the data when the sort button is pressed', () => {
    wrapper.simulate('click');
    expect(changeSortOrder).toHaveBeenCalledTimes(1);
    expect(reverseSortOrder).toHaveBeenCalledTimes(0);
  });

  it('Reverse orders the data when the sort button is pressed for the current sort', () => {
    const matchingProps = { ...componentProps, orderBy: 'test' };
    const matchedwrapper = shallow(
      <SortGlyph {...matchingProps} />,
    );
    matchedwrapper.simulate('click');
    expect(changeSortOrder).toHaveBeenCalledTimes(0);
    expect(reverseSortOrder).toHaveBeenCalledTimes(1);
  });
});
