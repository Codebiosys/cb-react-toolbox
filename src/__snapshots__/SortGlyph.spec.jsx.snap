// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Sort Glyph renders 1`] = `
<Glyphicon
  bsClass="glyphicon"
  className=""
  glyph="sort-by-alphabet"
  onClick={[Function]}
/>
`;
